src/data/fin/tuple/nat_antidiagonal.lean

```diff
@@ -0,0 +1,211 @@
+    { simp },
+    { simp [eq_comm] }, },
+  { refine fin.cons_induction (λ x₀ x, _) x,
+    have : (0 : fin k.succ) ∉ finset.image fin.succ (finset.univ : finset (fin k)) := by simp,
+    simp_rw [antidiagonal_tuple, list.mem_bind, list.mem_map, list.nat.mem_antidiagonal,
+      fin.univ_succ, finset.sum_insert this, fin.cons_zero,
+      finset.sum_image (λ x hx y hy h, fin.succ_injective _ h), fin.cons_succ, fin.cons_eq_cons,
+      exists_eq_right_right, ih, prod.exists],
+    split,
+    { rintros ⟨a, b, rfl, rfl, rfl⟩, refl },
+    { rintro rfl, exact ⟨_, _, rfl, rfl, rfl⟩, } },
+end
+
+/-- The antidiagonal of `n` does not contain duplicate entries. -/
+lemma nodup_antidiagonal_tuple (k n : ℕ) : list.nodup (antidiagonal_tuple k n) :=
+begin
+  induction k with k ih generalizing n,
+  { cases n,
+    { simp },
+    { simp [eq_comm] }, },
+  simp_rw [antidiagonal_tuple, list.nodup_bind],
+  split,
+  { intros i hi,
+    exact (ih i.snd).map (fin.cons_right_injective (i.fst : (λ _, ℕ) 0)), },
+  induction n,
+  { exact list.pairwise_singleton _ _ },
+  { rw list.nat.antidiagonal_succ,
+    refine list.pairwise.cons (λ a ha x hx₁ hx₂, _) (n_ih.map _ (λ a b h x hx₁ hx₂, _)),
+    { rw list.mem_map at hx₁ hx₂ ha,
+      obtain ⟨⟨a, -, rfl⟩, ⟨x₁, -, rfl⟩, ⟨x₂, -, h⟩⟩ := ⟨ha, hx₁, hx₂⟩,
+      rw fin.cons_eq_cons at h,
+      injection h.1, },
+    { rw list.mem_map at hx₁ hx₂,
+      obtain ⟨⟨x₁, hx₁, rfl⟩, ⟨x₂, hx₂, h₁₂⟩⟩ := ⟨hx₁, hx₂⟩,
+      dsimp at h₁₂,
+      rw [fin.cons_eq_cons, nat.succ_inj'] at h₁₂,
+      obtain ⟨h₁₂, rfl⟩ := h₁₂,
+      rw h₁₂ at h,
+      exact h (list.mem_map_of_mem _ hx₁) (list.mem_map_of_mem _ hx₂) }, },
+end
+
+@[simp] lemma antidiagonal_tuple_one (n : ℕ) : antidiagonal_tuple 1 n = [![n]] :=
+begin
+  simp_rw [antidiagonal_tuple, antidiagonal, list.range_succ, list.map_append, list.map_singleton,
+    tsub_self, list.bind_append, list.bind_singleton, antidiagonal_tuple_zero_zero,
+    list.map_singleton, list.map_bind],
+  conv_rhs { rw ← list.nil_append [![n]]},
+  congr' 1,
+  simp_rw [list.bind_eq_nil, list.mem_range, list.map_eq_nil],
+  intros x hx,
+  obtain ⟨m, rfl⟩ := nat.exists_eq_add_of_lt hx,
+  rw [add_assoc, add_tsub_cancel_left, antidiagonal_tuple_zero_succ],
+end
+
+lemma antidiagonal_tuple_two (n : ℕ) :
+  antidiagonal_tuple 2 n = (antidiagonal n).map (λ i, ![i.1, i.2]) :=
+begin
+  rw antidiagonal_tuple,
+  simp_rw [antidiagonal_tuple_one, list.map_singleton],
+  rw [list.map_eq_bind],
+  refl,
+end
+
+
+-- lemma antidiagonal_tuple_pairwise_lex (k n : ℕ) :
+--   (antidiagonal_tuple k n).pairwise (pi.lex (≤) (λ _, (≤))) :=
+-- begin
+--   induction k with k ih generalizing n,
+--   { cases n,
+--     { simp },
+--     { simp [eq_comm] }, },
+--   simp_rw [antidiagonal_tuple, list.pairwise_bind, list.pairwise_map, list.mem_map,
+--     forall_exists_index, and_imp, forall_apply_eq_imp_iff₂],
+--   simp only [mem_antidiagonal, prod.forall, and_imp, forall_apply_eq_imp_iff₂],
+--   split,
+-- end
```

Comment:
I lost interest in proving this - we don't seem to have an analogous result for `list.nat.antidiagonal` anyway

Reply:
Done in #13339, it wasn't too bad